### PR TITLE
feat: pytest integration

### PR DESCRIPTION
**Pull request by jirikuncar**
## Highlights

- **Built-in pytest plugin** that creates hierarchical trace spans for test sessions and individual tests
- **Zero-config CI integration** - automatically enables when `CI=true` and `LOGFIRE_TOKEN` is set
- **Flexible activation** via `--logfire` CLI flag, `pyproject.toml` config, or auto-enable in CI environments
- **Comprehensive test instrumentation** with rich attributes including test outcomes, durations, parameters, and code locations
- **Seamless integration** with existing Logfire instrumentation - all spans from instrumented libraries nest under test spans
- **W3C Trace Context support** for linking test traces to external distributed traces via `TRACEPARENT`

## Example Usage

```bash
# Enable via CLI
pytest --logfire

# Or configure in pyproject.toml
[tool.pytest]
logfire = true

# Auto-enables in CI with `LOGFIRE_TOKEN` set

### CODE REVIEW

**Comment by Kludex** [comment]
Nice. I want to try it.

File: docs/integrations/pytest.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+[tool.pytest.ini_options]
+logfire = true
+```
+
+Or `pytest.ini`:
+
+```ini
+[pytest]
+logfire = true
+```
```
Comment (Viicos) [inline docs/integrations/pytest.md]: Can we use [tabs](https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#content-tabs), and use the new configuration format from pytest 9.0? https://docs.pytest.org/en/stable/reference/customize.html#configuration-file-formats

I'd say one tab for `pytest.toml`, one for `pyproject.toml` (but not using the legacy `ini_options`), and one for `pytest.ini`

File: docs/integrations/pytest.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+### Auto-Enable in CI
+
+The plugin automatically enables when:
```
Comment (Viicos) [inline docs/integrations/pytest.md]:
```suggestion
The plugin automatically enables when both:
```

File: docs/integrations/pytest.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+- `CI` environment variable is set to `true` or `1`
```
Comment (Viicos) [inline docs/integrations/pytest.md]:
```suggestion
- `CI` environment variable is set to an non-empty string
```

File: docs/integrations/pytest.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+To explicitly disable the plugin (useful to override auto-enable in CI):
+
+```bash
+pytest --no-logfire
+```
+
+## Configuration Options
+
+| Option | CLI Flag | INI Option | Environment Variable | Default | Description |
```
Comment (Viicos) [inline docs/integrations/pytest.md]:
```suggestion
| Option | CLI Flag | Config option | Environment variable | Default | Description |
```

File: logfire/_internal/integrations/pytest.py [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+            if isinstance(report.longrepr, tuple) and len(report.longrepr) > 2:
+                skip_reason = str(report.longrepr[2])
+            else:  # pragma: no cover
+                # Edge case: longrepr is not a tuple or has unexpected structure
+                skip_reason = str(report.longrepr)
+        span.set_attribute('test.skip_reason', skip_reason)
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_setup(item: pytest.Item) -> Generator[None, Any, None]:
```
Comment (Viicos) [inline logfire/_internal/integrations/pytest.py]:
```suggestion
def pytest_runtest_setup(item: pytest.Item) -> Generator[None]:
```

File: logfire/_internal/integrations/pytest.py [hunk 6]
```diff
@@ -0,0 +1,563 @@
...
+        return
+
+    logfire_instance = plugin_config.logfire_instance
+
+    with logfire_instance.span('setup: {nodeid}', nodeid=item.nodeid):
+        yield
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_call(item: pytest.Item) -> Generator[None, Any, None]:
```
Comment (Viicos) [inline logfire/_internal/integrations/pytest.py]:
```suggestion
def pytest_runtest_call(item: pytest.Item) -> Generator[None]:
```

File: logfire/_internal/integrations/pytest.py [hunk 7]
```diff
@@ -0,0 +1,563 @@
...
+        return
+
+    logfire_instance = plugin_config.logfire_instance
+
+    with logfire_instance.span('call: {nodeid}', nodeid=item.nodeid):
+        yield
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_teardown(item: pytest.Item) -> Generator[None, Any, None]:
```
Comment (Viicos) [inline logfire/_internal/integrations/pytest.py]:
```suggestion
def pytest_runtest_teardown(item: pytest.Item) -> Generator[None]:
```

File: logfire/_internal/integrations/pytest.py [hunk 8]
```diff
@@ -0,0 +1,563 @@
...
+
+    span.__enter__()
+    session.stash[_SESSION_SPAN_KEY] = span
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_protocol(
+    item: pytest.Item,
+    nextitem: pytest.Item | None,
+) -> Generator[None, Any, None]:
```
Comment (Viicos) [inline logfire/_internal/integrations/pytest.py]: We can leverage type var defaults:

```suggestion
) -> Generator[None]::
```
